fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/algebra/TopN.java

```diff
@@ -0,0 +1,28 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package org.apache.doris.nereids.trees.plans.algebra;
+
+/**
+ * Common interface for logical/physical TopN.
+ */
+public interface TopN extends Sort {
```

Comment:
We already have limit and sort. Why we need TopN? What is the difference between TopN and a combination of sort and limit?

Reply:
yes, we will translate limit(sort) to topN when exploration. and look up witch has best cost